TestDbApiHook::test_run_no_log  randomly fails

Sometimes TestDbApiHook::test_run_no_log  randomly fails (another caplog case) - usually in compat tests:

```python
________________________ TestDbApiHook.test_run_no_log _________________________
providers/tests/common/sql/hooks/test_dbapi.py:516: in test_run_no_log
    assert len(caplog.messages) == 1
E   assert 2 == 1
E    +  where 2 = len(['Rows affected: 0', "Task was destroyed but it is pending!\ntask: <Task pending name='Task-113' coro=<<async_generator_asend without __name__>()> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f8d90c493a0>()]>>"])
E    +    where ['Rows affected: 0', "Task was destroyed but it is pending!\ntask: <Task pending name='Task-113' coro=<<async_generator_asend without __name__>()> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f8d90c493a0>()]>>"] = <_pytest.logging.LogCaptureFixture object at 0x7f8d72fc7cd0>.messages
----------------------------- Captured stdout call -----------------------------
```

For example : https://github.com/apache/airflow/actions/runs/12835627081/job/35795783211#step:11:2417

--- a/providers/common/sql/tests/unit/common/sql/hooks/test_dbapi.py
+++ b/providers/common/sql/tests/unit/common/sql/hooks/test_dbapi.py
@@ -539,7 +539,7 @@ class TestDbApiHook:
         statement = "SQL"
         caplog.clear()
         self.db_hook_no_log_sql.run(statement)
-        assert len(caplog.messages) in [1, 2]
+        assert "Rows affected: 0" in caplog.text
 
     def test_run_with_handler(self):
         sql = "SQL"

